src/converse-autocomplete.js

```diff
@@ -22,12 +22,13 @@ export const FILTER_STARTSWITH = function (text, input) {
 };
 
 
-const SORT_BYLENGTH = function (a, b) {
-    if (a.length !== b.length) {
-        return a.length - b.length;
+const SORT_BY_QUERY = function (a, b) {
+    const query = a.query;
+    if (a.label.indexOf(query) === b.label.indexOf(query)) {
+        return 0
```

Comment:
Instead of returning `0`, you can return `SORT_BY_LENGTH(a, b)` so that the longest relative match is shown higher.